Lib/mimetypes.py

```diff
@@ -599,47 +599,21 @@ def _default_mime_types():
 
 
 def _main():
-    import getopt
-
-    USAGE = """\
-Usage: mimetypes.py [options] type
-
-Options:
-    --help / -h       -- print this message and exit
-    --lenient / -l    -- additionally search of some common, but non-standard
-                         types.
-    --extension / -e  -- guess extension instead of type
-
-More than one type argument may be given.
-"""
-
-    def usage(code, msg=''):
-        print(USAGE)
-        if msg: print(msg)
-        sys.exit(code)
-
-    try:
-        opts, args = getopt.getopt(sys.argv[1:], 'hle',
-                                   ['help', 'lenient', 'extension'])
-    except getopt.error as msg:
-        usage(1, msg)
-
-    strict = 1
-    extension = 0
-    for opt, arg in opts:
-        if opt in ('-h', '--help'):
-            usage(0)
-        elif opt in ('-l', '--lenient'):
-            strict = 0
-        elif opt in ('-e', '--extension'):
-            extension = 1
-    for gtype in args:
-        if extension:
-            guess = guess_extension(gtype, strict)
+    from argparse import ArgumentParser
+    parser = ArgumentParser(description='compare directories and files')
+    parser.add_argument('-e', '--extension', action='store_true',
+                        help='guess extension instead of type')
+    parser.add_argument('-l', '--lenient', action='store_true',
+                        help='search also for common but non-standard types')
+    parser.add_argument('type', nargs='+', help='type to search')
+    arguments = parser.parse_args()
+    for gtype in arguments.type:
+        if arguments.extension:
+            guess = guess_extension(gtype, not arguments.lenient)
             if not guess: print("I don't know anything about type", gtype)
             else: print(guess)
         else:
-            guess, encoding = guess_type(gtype, strict)
+            guess, encoding = guess_type(gtype, not arguments.lenient)
             if not guess: print("I don't know anything about type", gtype)
             else: print('type:', guess, 'encoding:', encoding)
```

Comment:
You could reformat these as indented blocks rather than `conditional: expression`, but perhaps minor.